python/metatensor-torch/tests/operations/block_from_array.py

```diff
@@ -1,11 +1,12 @@
+import pytest
 import torch
 from packaging import version
 
 import metatensor.torch
 
 
-def check_operation(block_from_array):
-    values = torch.arange(42).reshape(2, 3, 7).to(torch.float64)
+def check_operation(block_from_array, device):
```

Comment:
just to be thorough, could you add a test in metatensor-core that creates a block with a `meta` torch tensor, and check that the labels are still on CPU?